### ISSUE TITLE
Firebase Perf is not logged on preview versions

### ISSUE THREAD

**Issue by jcantin-nventive** [xamarin/GooglePlayServicesComponents#228]
### Xamarin.Android Version (eg: 6.0):
Target SDK: Android 9.0 (Pie)

### Operating System & Version (eg: Mac OSX 10.11):
Android 6.0

### Google Play Services Version (eg: 8.4.0):
71.1620.0-preview9
We can see the issue appear as early as version 71.1620.0-preview3

### Describe your Issue:
Performance is not logged.

### Steps to Reproduce (with link to sample solution if possible):
[archive_20190702.zip](https://github.com/xamarin/GooglePlayServicesComponents/files/3351467/archive_20190702.zip)
Expand the included .zip sample solution
Add the missing google-services.json file, which you download from your Firebase project
Launch the firebaseperfunoexperiment.Droid project on an Android device

### Include any relevant Exception Stack traces, build logs, adb logs:
In the logs, we can see the following error:

 06-28 15:29:31.758 D/FirebasePerformance(20100): onResume(): md5b48014bba4061e79cb3bc92284d1786c.MainActivity: 4015271 microseconds
06-28 15:29:31.764 I/FA      (20100): Tag Manager is not found and thus will not be used
**06-28 15:29:31.774 W/FirebasePerformance(20100): App Instance ID is null or empty, dropping the log**

**Comment by MatFillion** [comment]
To add on this issue we wanted to provide another sample.

### Sample based on Xamarin.Forms
[perf_preview.zip](https://github.com/xamarin/GooglePlayServicesComponents/files/3356823/perf_preview.zip)
It is using 71.1620.0-preview9
Compiling on Android Pie (API 28)

### Steps to reproduce
Open the .sln attached
Add the missing google-services.json file, which you download from your Firebase project
Launch the android project on an Android device
I ran it on a Pixel 3 running Pie but it doesn't seem device related as we observe the same result on various devices (and OS versions)

### Include any relevant Exception Stack traces, build logs, adb logs:
[perf_logs_no_instance_id.txt](https://github.com/xamarin/GooglePlayServicesComponents/files/3356870/perf_logs_no_instance_id.txt)

Because of issue #172 we are unable to trace network calls on the stable version, so we're trying the preview versions but this issue comes into play. So we seem blocked on both paths we try.

If you need more information from us please don't hesitate.

**Comment by moljac** [comment]
Related #172

**Comment by moljac** [comment]
@jcantin-nventive 

Uno seems to need NDK and thus complicates our CI builds, so I moved your sample to my repository:

https://github.com/moljac/Samples.GooglePlayServices-n-Firebase 

I will setup builds there, but this repo will be submodule of it.

**Comment by moljac** [comment]
Please try release 71 and open new issue.

If necessary link to this issue please.

Closing.

**Comment by MatFillion** [comment]
We tried the same samples with the package https://www.nuget.org/packages/Xamarin.Firebase.Perf/71.1620.0 (and all other related packages)

We still get the same instance ID problems and no logs are sent (both startup logs and httpmetric logs)

@moljac is there a configuration step we would be missing or only updating the nuget packages should do the trick?


#### relevant logs:
Time    Device Name    Type    PID    Tag    Message
07-19 15:17:37.753    Google Pixel 3 XL    Warning    13718    FirebasePerformance    App Instance ID is null or empty, dropping the log
07-19 15:17:37.750    Google Pixel 3 XL    Debug    13718    FirebasePerformance    onResume(): md53c3c97ca3cc6c3c3f4e4b90f47d7e05a.MainActivity: 771932 microseconds
07-19 15:17:37.678    Google Pixel 3 XL    Info    13718    FirebasePerformance    Session ID - 40977e22072f441e93716206bd927fcc
07-19 15:17:37.151    Google Pixel 3 XL    Debug    13718    FirebasePerformance    Creating a new Non Verbose Session: 40977e22072f441e93716206bd927fcc
07-19 15:17:37.150    Google Pixel 3 XL    Debug    13718    FirebasePerformance    Creating a new Non Verbose Session: cd01f7f697ad4009a420a13e09fa3b9b
07-19 15:06:10.022    Google Pixel 3 XL    Warning    13428    FirebasePerformance    App Instance ID is null or empty, dropping the log
07-19 15:06:10.022    Google Pixel 3 XL    Debug    13428    FirebasePerformance    Logging TraceMetric - _as 13142ms
07-19 15:06:10.019    Google Pixel 3 XL    Debug    13428    FirebasePerformance    onResume(): md5563b6af7023eaa6da10c5306c2845b23.MainActivity: 13142732 microseconds
07-19 15:06:10.016    Google Pixel 3 XL    Info    13428    FirebasePerformance    Session ID - d6738e72808944d8ad48426cc47c2c13
07-19 15:06:09.456    Google Pixel 3 XL    Debug    13428    FirebasePerformance    Creating a new Non Verbose Session: d6738e72808944d8ad48426cc47c2c13
07-19 15:06:09.455    Google Pixel 3 XL    Debug    13428    FirebasePerformance    Creating a new Non Verbose Session: ec24c0f0cb524c4e91757526b2a3eefa

**Comment by MatFillion** [comment]
@moljac I created a new issue #234 with a repro sample.

**Comment by moljac** [comment]
@MatFillion Can you try ManifestMerger, please?
